linode/terraform-provider-linode
# new: Support control plane ACL in lke cluster resoure and datasource

## 📝 Description

Support control plane ACL CRUD operations in cluster resource and data source. 

## ✔️ How to Test

```
make PKG_NAME="linode/lke" int-test
```

Manual test:

1. In a sandbox environment, i.e. dx-dvenv, run the following tf config to create an LKE cluster with control plane ACL:
```
resource "linode_lke_cluster" "test" {
    label       = "my-cluster"     
    k8s_version = "1.28"           
    region      = "us-east"     
    tags        = ["prod"]         
    control_plane {
        acl {
            enabled = true
            addresses {
                ipv4 = "0.0.0.0/0"
                ipv6 = "2001:db8::/32"
            }
        }
    }
    pool {
        type  = "g6-standard-2"
        count = 1
    }
}
```
2. Run the above plan and observe the resource created successfully.
3. Disable the ACL
```
resource "linode_lke_cluster" "test" {
    label       = "my-cluster"     
    k8s_version = "1.28"           
    region      = "us-east"     
    tags        = ["prod"]         
    control_plane {
        acl {
            enabled = false
        }
    }
    pool {
        type  = "g6-standard-2"
        count = 1
    }
}
```
5. Read the data source from the updated LKE
```
data "linode_lke_cluster" "test" {
    id = linode_lke_cluster.test.id
}
```
6. Change around the configs to see if there is any edge case
